qiskit/dagcircuit/dagcircuit.py

```diff
@@ -491,8 +491,18 @@ def _add_op_node(self, op, qargs, cargs):
         self._increment_op(op)
         return node_index
 
-    def _copy_circuit_metadata(self):
-        """Return a copy of source_dag with metadata but empty."""
+    def copy_empty_like(self):
```

Comment:
If we're changing the name here I wonder if it makes sense to add an alias at the old name and raise a deprecation warning. While it wasn't public as you say in the commit message summary line it was widely used. So having something like:

```python
def _copy_circuit_metadata(self):
    warnings.warn("deprecated", DeprecationWarning)
    return self.copy_empty_like()
```

isn't too terrible to keep around for a release or two. Since it was private I think forgoing the requirement that the alternative/replacement is available in a release prior to the deprecation release (`2a` in: https://qiskit.org/documentation/contributing_to_qiskit.html#deprecation-policy ) is probably ok.

Reply:
I agree. Added in aaaffaa19c1aa656a40fa720cc2c4e6ad9bd5d69